src/components/Api.js

```diff
@@ -0,0 +1,104 @@
+export default class Api {
+  constructor(options) {
+    this.baseUrl = options.baseUrl;
+    this.headers = options.headers;
+  }
+
+  _checkResponse(response) {
+    if (!response.ok) {
+      return Promise.reject(`Error: ${response.status}`);
+    }
+    return response.json();
+  }
+
+  getAllData() {
+    return Promise.all([this.loadUserInfo(), this.getInitialCards()])
+      .then(([user, cards]) => {
+        return { user, cards };
+      })
+      .catch((error) => console.error("Error loading data:", error));
```

Comment:
It’s better not to use `catch` in `Api` because you need to `catch` possible errors at the very **end** of any request and this end is in `index.js`    after all blocks `then`. And in real projects blocks `catch` usually open error popups and so on.